peachjam/models/core_document_model.py

```diff
@@ -711,13 +711,10 @@ class Meta:
         abstract = True
 
     def save(self, *args, **kwargs):
-        if not self.filename:
-            self.filename = self.file.name
-        if not self.size:
-            self.size = self.file.size
-        if not self.mimetype:
-            self.file.seek(0)
-            self.mimetype = magic.from_buffer(self.file.read(), mime=True)
```

Comment:
this prevents us from updating the source file details

Reply:
This messes with setting these fields explicitly. Instead, clear them in the admin form so that they get reset here.